netbout-web/src/main/java/com/netbout/rest/Markdown.java

```diff
@@ -57,6 +58,24 @@
     private static final Tidy TIDY = Markdown.makeTidy();
 
     /**
+     * Plain link detection pattern.
+     * The regex consists of the following parts:
+     * <ul>
+     * <li>Negative lookbehind to determine we skip properly formatted
+     * links (e.g. [foo](http://bar))</li>
+     * <li>Scheme prefix followed by arbitrary number of URL-valid
+     * characters</li>
+     * <li>Last character &ndash; URL-valid characters with exclusion of common
+     * punctuation like dot, comma, question mark, exclamation mark, semicolon,
+     * colon, and parentheses</li>
+     * </ul>
+     */
+    private static final Pattern LINK = Pattern.compile(
+        // @checkstyle LineLength (1 line)
+        "(?<!(\\]\\())(https?://[a-zA-Z0-9-._~:/\\?#\\[\\]\\(\\)@!$&'*+,;=%]+[a-zA-Z0-9-_~/#\\[\\]@$&'*+=%])"
```

Comment:
@komaz can't we simplify this regex with something like:

`(?<!(\\]\\())(https?://.*)`

I mean: what's the worse thing that could happen if we do this? This regex seems very complex to me...